src/es/jkanime/src/eu/kanade/tachiyomi/animeextension/es/jkanime/Jkanime.kt

```diff
@@ -114,33 +124,64 @@ class Jkanime : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
+        val jsPath = scriptServers.data().substringAfter("= remote+'").substringBefore("'")
+        if (jsServer.isNotEmpty() && jsPath.isNotEmpty()) {
+            val jsLinks = client.newCall(GET(jsServer + jsPath)).execute().body.string()
+                .substringAfter("var servers = ").parseAs<Array<JsLinks>>().map {
+                    Pair(String(Base64.decode(it.remote, Base64.DEFAULT)), "${it.lang}".getLang())
+                }
+            servers.addAll(jsLinks)
+        }
+
+        val htmlLinks = document.select("div.col-lg-12.rounded.bg-servers.text-white.p-3.mt-2 a").map {
             val serverId = it.attr("data-id")
-            val langClass = it.attr("class")
-            val lang = if (langClass.contains("lg_3")) "[LAT]" else if (langClass.contains("lg_1")) "[JAP]" else ""
-            val scriptServers = document.selectFirst("script:containsData(var video = [];)")!!
-            val url = scriptServers.data().substringAfter("video[$serverId] = '<iframe class=\"player_conte\" src=\"")
+            val lang = it.attr("class").substringAfter("lg_").substringBefore(" ").getLang()
+            val url = scriptServers.data()
+                .substringAfter("video[$serverId] = '<iframe class=\"player_conte\" src=\"")
                 .substringBefore("\"")
                 .replace("/jkokru.php?u=", "http://ok.ru/videoembed/")
                 .replace("/jkvmixdrop.php?u=", "https://mixdrop.ag/e/")
                 .replace("/jksw.php?u=", "https://sfastwish.com/e/")
                 .replace("/jk.php?u=", "$baseUrl/")
+            Pair(if (url.contains("um2.php") || url.contains("um.php")) baseUrl + url else url, lang)
+        }
+        servers.addAll(htmlLinks)
+        return servers
+    }
 
-            try {
-                when {
-                    "ok" in url -> OkruExtractor(client).videosFromUrl(url, "$lang ").forEach { videos.add(it) }
-                    "mixdrop" in url -> MixDropExtractor(client).videosFromUrl(url, prefix = "$lang ").forEach { videos.add(it) }
-                    "sfastwish" in url -> StreamWishExtractor(client, headers).videosFromUrl(url, prefix = "$lang StreamWish").forEach { videos.add(it) }
-                    "stream/jkmedia" in url -> videos.add(Video(url, "$lang Xtreme S", url))
-                    "um2.php" in url -> JkanimeExtractor(client).getNozomiFromUrl(baseUrl + url, "$lang ").let { if (it != null) videos.add(it) }
-                    "um.php" in url -> JkanimeExtractor(client).getDesuFromUrl(baseUrl + url, "$lang ").let { if (it != null) videos.add(it) }
-                }
-            } catch (_: Exception) {}
+    override fun videoListParse(response: Response): List<Video> {
+        val document = response.asJsoup()
+        return getVideoLinks(document).parallelCatchingFlatMapBlocking { (url, lang) ->
+            when {
+                "ok" in url -> OkruExtractor(client).videosFromUrl(url, "$lang ")
```

Comment:
Move the extractors out to a class instance so they don't get initialized every time